src/components/CreateProposal/CreateProposalContainer.tsx

```diff
@@ -159,125 +117,189 @@ class CreateProposalContainer extends React.Component<IProps, IState> {
-            placeholder="Summarize your propsoal"
-            ref="titleNode"
-            required
-            type="text"
-            value={this.state.title}
-          />
-          <label htmlFor="descriptionInput">
-            Description (URL)
-            <a className={css.recommendedTemplate} href="https://docs.google.com/document/d/1JzBUikOfEll9gaJ9N2OfaJJeelWxoHzffNcZqeqWDTM/edit#heading=h.vaikfqc64l1" target="_blank">
-              Recommended template
-            </a>
-            <img className={css.infoTooltip} src="/assets/images/Icon/Info.svg"/>
-          </label>
-          <input
-            id="descriptionInput"
-            onChange={this.handleChange}
-            placeholder="Proposal description URL"
-            ref="descriptionNode"
-            required
-            type="text"
-            value={this.state.description}
-          />
-          <label htmlFor="beneficiaryInput">
-            Target Address
-            <img className={css.infoTooltip} src="/assets/images/Icon/Info.svg"/>
-          </label>
-          <input
-            id="beneficiaryInput"
-            className={this.state.errors.beneficiaryAddress ? css.error : null}
-            maxLength={42}
-            onChange={this.handleChange}
-            onBlur={async (e) => await this.validate()}
-            placeholder="Recipient's address public key"
-            ref="beneficiaryNode"
-            required
-            type="text"
-            value={this.state.beneficiaryAddress}
-          />
-          {this.state.errors.beneficiaryAddress ?
-            <span className={css.errorMessage}>
-              {this.state.errors.beneficiaryAddress}
-            </span>
-            : ""
-          }
-          <div className={css.addTransfer}>
-            <div style={{display: 'none'}}>
-              <label htmlFor="nativeTokenRewardInput">{dao.tokenSymbol} reward: </label>
-              <input
-                id="nativeTokenRewardInput"
-                maxLength={10}
-                onChange={this.handleChange}
-                placeholder="How many tokens to reward"
-                ref="nativeTokenRewardNode"
-                // required
-                type="number"
-                value={this.state.nativeTokenReward}
+        <Formik
+          initialValues={{
+            beneficiaryAddress: '',
+            description: '',
+            ethReward: 0,
+            externalTokenReward: 0,
+            nativeTokenReward: 0,
+            reputationReward: 0,
+            title: ''
+          } as FormValues}
+          validate={(values: FormValues) => {
+            const {
+              beneficiaryAddress,
+              description,
+              ethReward,
+              externalTokenReward,
+              nativeTokenReward,
+              reputationReward,
+              title
+            } = values;
+            const errors: any = {};
+
+            const require = (name: string) => {
```

Comment:
seems really weird that we have to define require here. so we have to do this for every form we want to validate?

Reply:
just for convenience really... Formik doesn't provide built in validation rules. We can use HTML5's `required` but the error message would look different than the others. 
If it gets used more than once we'll extract it to a separate module.